Mathlib/RingTheory/Radical.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2024 Jineon Back and Seewoo Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jineon Baek, Seewoo Lee
+-/
+import Mathlib.RingTheory.UniqueFactorizationDomain
+
+/-!
+# Radical of an element of a unique factorization normalization monoid
+
+This file defines a radical of an element `a` of a unique factorization normalization
+monoid, which is defined as a product of normalized prime factors of `a` without duplication.
+This is different from the radical of an ideal.
+
+## Main declarations
+
+- `radical`: The radical of an element `a` in a unique factorization monoid is the product of
+  its prime factors.
+- `radical_eq_of_associated`: If `a` and `b` are associates, i.e. `a * u = b` for some unit `u`,
+  then `radical a = radical b`.
+- `radical_unit_mul`: Multiplying unit does not change the radical.
+- `radical_dvd_self`: `radical a` divides `a`.
+- `radical_pow`: `radical (a ^ n) = radical a` for any `n ≥ 1`
+- `radical_of_prime`: Radical of a prime element is equal to its normalization
+- `radical_pow_of_prime`: Radical of a power of prime element is equal to its normalization
+
+## TODO
+
+- Make a comparison with `Ideal.radical`. Especially, for principal ideal,
+  `Ideal.radical (Ideal.span {a}) = Ideal.span {radical a}`.
+- Prove `radical (radical a) = radical a`.
+-/
+
+noncomputable section
+
+open scoped Classical
+
+open UniqueFactorizationMonoid
+
+-- `CancelCommMonoidWithZero` is required by `UniqueFactorizationMonoid`
+variable {M : Type*} [CancelCommMonoidWithZero M] [NormalizationMonoid M]
+  [UniqueFactorizationMonoid M]
+
+/-- The finite set of prime factors of an element in a unique factorization monoid. -/
+def primeFactors (a : M) : Finset M :=
+  (normalizedFactors a).toFinset
+
+/--
+Radical of an element `a` in a unique factorization monoid is the product of
+the prime factors of `a`.
+-/
+def radical (a : M) : M :=
```

Comment:
Don't you think this is too generic of a name to be put in the root namespace?

Reply:
@YaelDillies I agree, do you have any suggestions? I was thinking something like `element_radical`. Maybe I should also change the name of the file `Radical.lean` accordingly?

Reply:
The defs you're using are in the `UniqueFactorizationDomain` namespace. m not a huge fan of it (why not just `UFD` or `RingTheory`?) but it would be an obvious statu quo choice

Reply:
I may not understand completely but moving everything to the `RingTheory/UniqueFactorizationDomain.lean` would be okay?

Reply:
No, I'm talking about the namespace, not the file!